plugin/iamutil/resource_parser_test.go

```diff
@@ -39,8 +39,10 @@ func TestEnabledIamResources_RelativeName(t *testing.T) {
 			}
 
 			if resource != nil {
-				if err = verifyResource(resourceType, resource.(*IamResource)); err != nil {
-					t.Errorf("could not verify resource for relative resource name %q: %sv", testRelName, err)
+				if r, ok := resource.(*IamResource); ok {
```

Comment:
I wonder if we should modify or add a new test that verifies `DatasetResource` from the `generatedResources` collection, or at least verify that `Parse` can return that object of that type to get code coverage on the fix.

Reply:
Good call. Looking at the existing coverage. Will add a test here if needed 👍

Reply:
Updated these tests to add coverage for both IAM and dataset resources in 9ce72d1. Used the Resource interface so that both can be used.